misk-hibernate/src/main/kotlin/misk/hibernate/SchemaMigrator.kt

```diff
@@ -60,59 +84,81 @@ internal class SchemaMigrator(
       sessionFactory.doWork {
         val statement = createStatement()
         statement.addBatch("""
-            |CREATE TABLE schema_version (
+            |CREATE TABLE schema_versions (
+            |  namespace varchar(191) NOT NULL,
             |  version varchar(50) NOT NULL,
             |  installed_by varchar(30) DEFAULT NULL,
-            |  UNIQUE KEY (version)
+            |  UNIQUE KEY (namespace, version)
             |);
             |""".trimMargin())
         statement.executeBatch()
       }
+
+      // Temporary migration to new schema - can be removed after all services upgraded.
```

Comment:
Yeah, it though it would be roughly similar, since it would have to be conditional alter, which has to be either a stored procedure or code like this.